### DISCUSSION TITLE
Flagging logs for highlighting

### DISCUSSION THREAD

**Discussion by Seggr** [datalust/seq-tickets#2042] [Ideas]
I would find it very helpful to flag specific logs to highlight with a color in the same manner that we add filters, as it can be hard sometimes to keep track of a single event when scrolling through endless events.

**Comment by KodrAus** [comment]
Thanks for the suggestion @Seggr :+1: Some kind of session-level flagging or pinning like you typically see in email clients could definitely be helpful here.

**Comment by Giddy-Up224** [comment]
Thanks @Seggr, this is my greatest frustration while using Seq. I love the power of Seq, but there' s no way to highlight specific words or lines and keep the other lines visible. This forces me to use Seq and Notepad++ so I can highlight specific words. Needless to say, it's not satisfactory.

Seq Developers, it would be AWESOME to have this feature! Thanks in advance!

**Comment by KodrAus** [comment]
That's a helpful datapoint, thanks @Giddy-Up224. We don't have any immediate plan to implement this just yet, but it's still on our radar.

**Reply by Giddy-Up224** [reply]
Thanks for the reply, @KodrAus. What must we users do to get this on your immediate plan? 😉

**Comment by cliffi30** [comment]
Hello, is there any poll or tracking where one can highlight such a feature? It comes up frequently from different seq users as it seems. I don't give up hope, yet, that is comes at some point in the future if we keep asking for it long enough. :) Thanks

**Comment by liammclennan** [comment]
I wonder if anyone could provide some more specifics on what they are trying to achieve? It seems there are several different features being discussed here. 

@Seggr's original request was to be able to highlight one event, which is easy, but limited, because if the search context changes then the event might not be visible. 

@Giddy-Up224 it sounds like you want to work with multiple result sets simultaneously, ala something like [Datadog Notebooks](https://docs.datadoghq.com/notebooks/). Is that correct?

**Reply by Giddy-Up224** [reply]
The features in [Datadog Notebooks](https://docs.datadoghq.com/notebooks/), aren't  really what I'm looking for. I'm looking for something as simple as the ability to apply different background colors to text depending on the "Matching Pattern". Here's an example from [Klogg](https://github.com/variar/klogg/).

![image](https://github.com/datalust/seq-tickets/assets/128929355/fe2a374c-83e0-4909-b4c1-b445b67639e4)

I've got it set to apply this background to the whole line, but you can also choose to only highlight the matching text.
It's called a Highlighter and is completely customizable. You can save multiple Highlighters and simply select the one you wish to use.

The highlighter feature allows me to see all events that I've filtered for and pick out the different _types_ of events by color. The _types_ in this case are determined simply by my Highlighter "Matching Pattern". This is very important when troubleshooting machinery movements etc.

@liammclennan, as stated in my first post, I like the power of Seq, but need this one capability to make it really useful. Without this feature, I'm simply looking at mind-numbing quantities of text with no way to categorize it visually. You're probably wondering why I don't just stick with Klogg? It has it's limitations. I cannot query logs in Klogg like I can in Seq. I cannot create aggregate data from multiple machines in Klogg.

I hope this explains it. If you have any more questions please feel free to ask. Thanks!

**Reply by liammclennan** [reply]
Thanks for the extra detail. I like Klogg's old school interface. :)

**Comment by sbloomfieldtea** [comment]
Here's an example from Hoo WinTail where I highlight trade and position with different background colors:
![image](https://github.com/user-attachments/assets/8cc000d3-12d7-4359-a550-82a225cbcfb3)

If it hasn't already been mentioned, aside from generic text highlights, perhaps a setting for specific log parameters, so if I log {TradeNumber} or {Position} I can set a color rule for those property names.

I tried a chrome extensions for setting highlights but it was difficult to work with, so a feature like this would be awesome.
